.github/workflows/ros2_build_test/Dockerfile

```diff
@@ -9,16 +9,16 @@ WORKDIR /root/ws/src
 COPY . src
 
 RUN <<deps
-  apt-get update
-  apt-get install -y libboost-all-dev
-  source /opt/ros/humble/setup.bash
-  rosdep update
+  apt-get update || exit 1
+  apt-get install -y libboost-all-dev || exit 1
+  source /opt/ros/humble/setup.bash || exit 1
```

Comment:
Since this is pretty verbose as is I think we can maybe skip for things like `source` that are unlikely to fail (and if they do fail, will almost certainly cause some later step to fail)